### PR TITLE
JAMES-2186 GetMessages PR

### PR DESCRIPTION
**Pull request by chibenwa**
Retrigger https://github.com/linagora/james-project/pull/1046

### CODE REVIEW

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/GetMessages.feature [hunk 1]
```diff
@@ -36,6 +36,24 @@ Feature: GetMessages method
...
+    And the id of the message is "m1"
+
+  Scenario: Retrieving a message in a mailbox not delegated to me
+    Given "alice@domain.tld" has a mailbox "notShared"
+    And the user has a message "m1" in "notShared" mailbox with subject "my test subject", content "testmail"
+    Given "bob@domain.tld" is connected
+    When the user ask for messages "m1"
+    Then no error is returned
+    And the list should contain 0 message
+
```
Comment (rouazana) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/GetMessages.feature]: Can you add:
Scenario: Retrieving a message in a mailbox delegated to someone else

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadGet.feature [hunk 2]
```diff
@@ -80,4 +81,22 @@ Feature: Download GET
     And a connected user "username1@domain.tld"
     And "username1@domain.tld" has a mailbox "INBOX"
     When "username1@domain.tld" downloads "1"
-    Then the user should receive a not found response
\ No newline at end of file
+    Then the user should receive a not found response
+
+  Scenario: User cannot download attachment of another user when shared mailbox
```
Comment (rouazana) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadGet.feature]: please add a fixup s/cannot/can to ed7f58342d292fcd4311ba7ff8d6b18fa51475f3

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadGet.feature [hunk 3]
```diff
@@ -80,4 +81,22 @@ Feature: Download GET
...
+    Given "username@domain.tld" mailbox "sharedMailbox" contains a message "1" with an attachment "2"
+    And "username@domain.tld" shares its mailbox "sharedMailbox" with "username1@domain.tld"
+    And a connected user "username1@domain.tld"
+    And "username1@domain.tld" has a mailbox "sharedMailbox"
+    When "username1@domain.tld" downloads "2"
+    Then the user should receive that blob
+    And the blob size is 3071
+
+  Scenario: User cannot download message blob of another user when shared mailbox
```
Comment (rouazana) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadGet.feature]: please add a fixup s/cannot/can to ed7f58342d292fcd4311ba7ff8d6b18fa51475f3

Comment (rouazana) [reply server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadGet.feature]: please add following scenarios:
Scenario: User cannot download attachment of another user when shared mailbox to someone else
Scenario: User cannot download message blob of another user when shared mailbox to someone else

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadGet.feature [hunk 4]
```diff
@@ -23,80 +23,80 @@ Feature: Download GET
...
+  Scenario: User can download attachment of another user when shared mailbox
+    Given "alice@domain.tld" mailbox "sharedMailbox" contains a message "1" with an attachment "2"
+    And "alice@domain.tld" shares its mailbox "sharedMailbox" with "bob@domain.tld"
+    And a connected user "bob@domain.tld"
+    And "bob@domain.tld" has a mailbox "sharedMailbox"
+    When "bob@domain.tld" downloads "2"
     Then the user should receive that blob
     And the blob size is 3071
 
-  Scenario: User cannot download message blob of another user when shared mailbox
```
Comment (rouazana) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadGet.feature]: this s/can/cannot should not be done in this commit 8443e97001d50990196737a399471773e9041165

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadGet.feature [hunk 5]
```diff
@@ -23,80 +23,80 @@ Feature: Download GET
...
-    Given "username@domain.tld" mailbox "sharedMailbox" contains a message "1" with an attachment "2"
-    And "username@domain.tld" shares its mailbox "sharedMailbox" with "username1@domain.tld"
-    And a connected user "username1@domain.tld"
-    And "username1@domain.tld" has a mailbox "sharedMailbox"
-    When "username1@domain.tld" downloads "1"
+  Scenario: User can download message blob of another user when shared mailbox
```
Comment (rouazana) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadGet.feature]: idem 8443e97001d50990196737a399471773e9041165

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadGet.feature (hunk 4 above)
Comment (chibenwa) [reply server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadGet.feature]: Done

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadGet.feature (hunk 5 above)
Comment (chibenwa) [reply server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadGet.feature]: Done

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadGet.feature (hunk 4 above)
Comment (chibenwa) [reply server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadGet.feature]: Done

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadGet.feature (hunk 5 above)
Comment (chibenwa) [reply server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadGet.feature]: Done

**Comment by mbaechler** [comment]
merged

**Pull request closed without merging**